File: tests/r/test_runner_cli.R
source(file.path("probes", "_runner_r.R"))

CAPABILITY <- "r_runner_cli_test"

args <- parse_args(CAPABILITY, character(0))
expected <- file.path("artifacts", paste0(CAPABILITY, ".json"))
if (!identical(args$output, expected)) {
  stop(sprintf("Expected default output '%s' but saw '%s'", expected, args$output), call. = FALSE)
}

exit_code <- persist_result(
  list(
    capability = CAPABILITY,
    status = "supported",
    detail = "R runtime emits JSON artifacts"
  ),
  args$output
)

if (exit_code != 0) {
  stop(sprintf("persist_result returned %d for supported status", exit_code), call. = FALSE)
}

if (!file.exists(args$output)) {
  stop(sprintf("Artifact '%s' was not created", args$output), call. = FALSE)
}

quit(status = 0, save = "no")

File: tests/c/test_runner_cli.c
#include "_runner_c.h"

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

static const char *CAPABILITY = "c_runner_cli_test";

int main(void) {
    const char *probe_id = "c_runner_cli_test_specimen";
    if (setenv("PROBE_ID", probe_id, 1) != 0) {
        fprintf(stderr, "Failed to set PROBE_ID\n");
        return 1;
    }

    struct probe_cli cli;
    if (probe_cli_init(&cli, CAPABILITY) != 0) {
        fprintf(stderr, "probe_cli_init failed\n");
        return 1;
    }

    char program_name[] = "test_runner_cli";
    char *argv[] = {program_name, NULL};
    if (probe_cli_parse(&cli, 1, argv) != 0) {
        fprintf(stderr, "probe_cli_parse rejected default arguments\n");
        return 1;
    }

    const char *expected = "artifacts/c_runner_cli_test_specimen.json";
    const char *resolved = probe_cli_output_path(&cli);
    if (resolved == NULL || strcmp(resolved, expected) != 0) {
        fprintf(stderr, "expected output path '%s' but saw '%s'\n", expected, resolved);
        return 1;
    }

    struct probe_result result = {
        .capability = CAPABILITY,
        .status = "supported",
        .detail = "c runtime emits JSON artifacts",
    };

    if (emit_result(&result, resolved) != 0) {
        fprintf(stderr, "emit_result returned failure\n");
        return 1;
    }

    if (access(resolved, F_OK) != 0) {
        fprintf(stderr, "Artifact '%s' does not exist\n", resolved);
        return 1;
    }

    return 0;
}
